qtpy/__init__.py

```diff
@@ -98,14 +98,13 @@ class PythonQtWarning(Warning):
 # Detecting if a binding was specified by the user
 binding_specified = QT_API in os.environ
 
-# Setting a default value for QT_API
-os.environ.setdefault(QT_API, 'pyqt5')
-
 API_NAMES = {'pyqt5': 'PyQt5', 'pyqt6': 'PyQt6',
              'pyside2':'PySide2', 'pyside6': 'PySide6'}
-API = os.environ[QT_API].lower()
+API = os.environ.get(QT_API, 'pyqt5').lower()
 initial_api = API
-assert API in API_NAMES
+if API not in API_NAMES:
+    raise ValueError(
```

Comment:
Maybe we could use here a custom exception like `PythonQtValueError(ValueError)`?